src/utils/packageHasher.js

```diff
@@ -0,0 +1,29 @@
+const logger = require('../Logger');
+const fs = require('fs');
+const md5File = require('md5-file');
+const walkedFiles = new Map();
+
+
+async function packageHasher(mainBundle, detailed = false) {
+  walkBundles(mainBundle, '--');
+
+  console.log(walkedFiles);
+}
+
+module.exports = packageHasher;
+
+async function walkBundles(bundle, indent) {
+  for (let child of bundle.childBundles) {
+    if ( !walkedFiles.has(child.name)) {
+      walkBundles(child, indent + '--')
+    }
+  }
+  hash = md5File.sync(bundle.name);
```

Comment:
Above this line, we need to do a find and replace on the current file (`bundle.name`) with the contents of `walkedFiles` (which has a map from old hashes to new hashes).